src/transactions/TransactionUtils.cpp

```diff
@@ -1539,15 +1541,24 @@ removeOffersAndPoolShareTrustLines(AbstractLedgerTxn& ltx,
 
         auto redeemIntoClaimableBalance =
             [&ltxInner, &txSourceID, txSeqNum, opIndex, &poolID, &accountID,
-             &cbSponsoringAccID](Asset const& assetInPool,
-                                 int64_t amount) -> RemoveResult {
+             &cbSponsoringAccID, &opEventManager](
+                Asset const& assetInPool, int64_t amount) -> RemoveResult {
             // if the amount is 0 or the claimant is the issuer, then we don't
             // create the claimable balance
-            if (isIssuer(accountID, assetInPool) || amount == 0)
+            if (amount == 0)
             {
                 return RemoveResult::SUCCESS;
             }
 
+            if (isIssuer(accountID, assetInPool))
+            {
+
+                opEventManager.newBurnEvent(
+                    assetInPool, liquidityPoolIDToSCAddress(poolID), amount);
+
+                return RemoveResult::SUCCESS;
+            }
```

Comment:
It would be nice to combine this event with the transfer event logic below with a single call to `eventForTransferWithIssuerCheck`. that would require moving this early return a few lines below, after the entry is constructed though, which maybe okay?